### ISSUE TITLE
[strict-boolean-expressions] False positive with union types having only one falsey value

### ISSUE THREAD

**Issue by tkalliom** [typescript-eslint/typescript-eslint#3115]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```javascript
module.exports = {
    "root": true,
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "tsconfigRootDir": __dirname,
        "project": ["./tsconfig.json"]
    },
    "plugins": ["@typescript-eslint"],
    "rules": {
        "@typescript-eslint/strict-boolean-expressions": ["error"]
    }
}
```

```TS
function f(arg: "Option1" | "Option2" | null) {
    if (arg)
        console.log(`Input: ${arg}`);
    else
        console.log("Null was input");
}
```

tsconfig: the default as output by `tsc --init`

**Expected Result**

I expected there to be no error, as there is exactly one possible falsey value, no ambiguity.

**Actual Result**
```
  2:9  error  Unexpected nullable string value in conditional. Please handle the nullish/empty cases explicitly  @typescript-eslint/strict-boolean-expressions
```
**Additional Info**
This is similar to the likely non-bug #2894, but in that example there are multiple falsey values.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.15.2` |
| `@typescript-eslint/parser`        | `4.15.2` |
| `TypeScript`                       | `4.1.5` |
| `ESLint`                           | `7.20.0` |
| `node`                             | `10.24.0` |
